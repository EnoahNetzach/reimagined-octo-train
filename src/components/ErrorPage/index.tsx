import { useTranslation } from 'react-i18next'
import { useRouteError } from 'react-router-dom'

export default function ErrorPage() {
  const { t } = useTranslation()
  const error = useRouteError()
  console.error(error)

  return (
    <main>
      <h1>{t('error.title')}</h1>

      <p>{t('error.body')}</p>
    </main>
  )
}
