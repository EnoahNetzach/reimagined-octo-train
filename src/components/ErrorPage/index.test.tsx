import { createRenderer } from 'react-test-renderer/shallow'
import ErrorPage from '.'

describe('ErrorPage', () => {
  it('mounts', () => {
    const consoleError = jest.spyOn(console, 'error').mockImplementationOnce(jest.fn())

    const renderer = createRenderer()
    renderer.render(<ErrorPage />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()

    expect(consoleError).toHaveBeenCalledWith(expect.any(Error))
  })
})
