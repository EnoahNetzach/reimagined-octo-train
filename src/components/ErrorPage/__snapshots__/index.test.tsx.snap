// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage mounts 1`] = `
<main>
  <h1>
    t(error.title)
  </h1>
  <p>
    t(error.body)
  </p>
</main>
`;
